### ISSUE TITLE
Standard tinting fills complete bounding box

### ISSUE THREAD

**Issue by PrimaryFeather** [zathras/jovial_svg#68]
First of all, thanks for your excellent library! You really did a fantastic job with it. 

However, I just tried out tinting, and somehow I can't get it to work. Here's a minimal code sample:

```dart
void main() {
  runApp(const JovialTestApp());
}

const svgString = '''
<svg version="1.1" width="50" height="50" xmlns="http://www.w3.org/2000/svg">
  <circle cx="25" cy="25" r="20" fill="green" />
</svg>
''';

class JovialTestApp extends StatelessWidget {
  const JovialTestApp({super.key});

  @override
  Widget build(BuildContext context) {
    final image = ScalableImage.fromSvgString(svgString);
    return MaterialApp(
      home: Scaffold(
        body: Row(children: [
          ScalableImageWidget(si: image),
          ScalableImageWidget(
              si: image.modifyTint(
                  newTintMode: BlendMode.srcIn, newTintColor: Colors.red)),
        ]),
      ),
    );
  }
}
```

As you can see, I'm displaying a simple SVG showing a circle two times; once just as it is (green), and subsequently with a red tint. I'm getting the following output:

<img width="206" alt="Screenshot 2023-08-07 at 21 51 57" src="https://github.com/zathras/jovial_svg/assets/547347/504238cc-8f90-4e94-a9df-026b480935b4">

Of course, I would expect a red circle instead of the red bounding box.

Maybe I'm just using this feature in the wrong way? Any help would be appreciated. 

**Additional context**
I've compiled and run the sample on macOS and iOS (in their most recent versions) and built with the latest Flutter (v3.10.6).

**Comment by billf-pshs** [comment]
Interesting!  You are absolutely correct.  Thanks for finding this.

`jovial_svg` was applying the tint to whatever was on the canvas, and not just the SVG.  So, if the canvas contained non-transparent pixels, the tint was being applied to those pixels too.  Since here the background was a white fill, every pixel was fully opaque, so the tint covered the entire extent of the SVG.

A call to `Canvas.saveLayer` is needed.

Tint is a feature from Android VectorDrawables, which is maybe why nobody noticed until now.  Tint is, of course, on the public API for `ScalableImage`, so it should be properly supported.

**Comment by PrimaryFeather** [comment]
Thank you so much for taking care of this so quickly, Bill!

Ah, yeah, I think I understand the problem now. Since the SVG was drawn into the same layer as the background, the color was drawn over the background pixels. Makes sense! 👍 

I applied your fix locally, and it works like a charm – as expected. :smile:
If there's anything I can do to help with testing, just let me know.

Again, your work on this is really appreciated! 🙏

**Comment by zathras** [comment]
Great!  Would you mind switching your pubspec over to 1.1.17-rc.1, just to make sure I didn't miss something dumb?

It's always nice when a bug report has the problem so well isolated, and is easy to reproduce.

**Comment by PrimaryFeather** [comment]
I just switched to `1.1.17-rc.1`, and that fixed the issue nicely! Plus, all my test SVGs are still rendering beautifully, so it doesn't _seem_ to have any side effects. 🎉

Well, I guess we have to thank the bug for being so easy to reproduce! 😆 
But I totally get what you mean, and I'm glad I could streamline the process for you. 😁

In any case: it's fixed! Awesome!!! Sending a huge THANK YOU from this side of the network. 🙏🎈

**Comment by billf-pshs** [comment]
Great, thanks for checking.  I'll probably publish 1.1.17 tonight, when I get back to the computer with those login credentials :-)

**Comment by zathras** [comment]
Published in 1.1.17
